screens/Map.js

```diff
@@ -1,31 +1,92 @@
 import React from 'react';
 import {
-  Image,
-  Platform,
-  ScrollView,
-  StyleSheet,
-  Text,
-  TouchableOpacity,
-  View,
-} from 'react-native';
-import { WebBrowser, MapView, Marker } from 'expo';
+  MapView,
+  Location,
+  Permissions,
+} from 'expo';
+
+import data from '../data.js'
 
 export default class Map extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      markers: [],
+      followsUserLocation: true,
+      mapRegion: {
+        latitude: 40.673868,
+        longitude: -73.970089,
+        latitudeDelta: 0.0922,
+        longitudeDelta: 0.0421,
+      },
+      locationResult: ""
+    };
+  };
+
   static navigationOptions = {
     header: null,
   };
 
+  componentDidMount = async () => {
+    this.setState({
+      markers: data
+    });
+    await this._getLocationAsync();
+  };
+
+  _getLocationAsync = async () => {
+    let {
+      status
+    } = await Permissions.askAsync(Permissions.LOCATION);
+    if (status !== 'granted') {
+      this.setState({
+        locationResult: 'Permission to access location was denied',
+        location,
+      });
+    }
+
+    let location = await Location.getCurrentPositionAsync({});
+ 
+    this.setState({
+      initialRegion: {
```

Comment:
i don't see initialRegion initiated on state, but maybe i'm missing something?

Reply:
ohhh it looks like you called it mapRegion up above and initialRegion here.

Reply:
I should have totally left a comment on this. So mapRegion is doing something different, that  changes when the user drags the map. That's what it makes it so that it doesn't recentralize back to the user's location after they stop dragged.

But you're right that initialRegion isn't initiated on state. There was something wonky going on with the map component so whenever I tried to initiate it, it would break and never register the setState that changes it to the user location. It would just stay on that initial empty object.

Reply:
oh that's suuuuuper weird. i'll go ahead and merge it in as is as and we can add a comment later.